compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmDefaultArgumentStubGenerator.kt

```diff
@@ -39,11 +38,7 @@ class JvmDefaultArgumentStubGenerator(override val context: JvmBackendContext) :
         //
         // This control flow limits us to an if-then (without an else), and this together with the
         // restriction on loading the parameter in the default case means we cannot create any temporaries.
-        +irIfThen(defaultFlag, irCall(this@JvmDefaultArgumentStubGenerator.context.ir.symbols.reassignParameterIntrinsic).apply {
-            putTypeArgument(0, parameter.type)
-            putValueArgument(0, irGet(parameter))
-            putValueArgument(1, default)
-        })
+        +irIfThen(defaultFlag, irSet(parameter.symbol, default))
```

Comment:
Real change: remove the JVM-specific intrinsic and use the IR structure instead.